src/common/api_utils.cpp

```diff
@@ -1,131 +1,23 @@
-	Aws::Http::URI uri;
-
-	// TODO move this to IRCatalog::GetBaseURL()
-	auto service = GetAwsService(endpoint_builder.GetHost());
-	auto region = GetAwsRegion(endpoint_builder.GetHost());
-
-	auto decomposed_host = DecomposeHost(endpoint_builder.GetHost());
-	for (auto &component : decomposed_host.path_components) {
-		uri.AddPathSegment(component);
-	}
-
-	for (auto &component : endpoint_builder.path_components) {
-		uri.AddPathSegment(component);
-	}
-
-	for (auto &param : endpoint_builder.GetParams()) {
-		const char *key = param.first.c_str();
-		auto value = param.second.c_str();
-		uri.AddQueryStringParameter(key, value);
-	}
-
-	Aws::Http::Scheme scheme = Aws::Http::Scheme::HTTPS;
-	uri.SetScheme(scheme);
-	// set host
-	uri.SetAuthority(decomposed_host.authority);
-
-	const Aws::Http::URI uri_const = Aws::Http::URI(uri);
-	auto create_http_req = Aws::Http::CreateHttpRequest(uri_const, Aws::Http::HttpMethod::HTTP_GET,
-	                                                    Aws::Utils::Stream::DefaultResponseStreamFactoryMethod);
-
-	std::shared_ptr<Aws::Http::HttpRequest> req(create_http_req);
-
-	// Set the user Agent.
-	auto &config = DBConfig::GetConfig(context);
-	req->SetUserAgent(config.UserAgent());
-
-	// will error if no secret can be found for AWS services
-	auto secret_entry = IRCatalog::GetStorageSecret(context, secret_name);
-	auto kv_secret = dynamic_cast<const KeyValueSecret &>(*secret_entry->secret);
-
-	std::shared_ptr<Aws::Auth::AWSCredentialsProviderChain> provider;
-	provider = std::make_shared<DuckDBSecretCredentialProvider>(
-	    kv_secret.secret_map["key_id"].GetValue<string>(), kv_secret.secret_map["secret"].GetValue<string>(),
-	    kv_secret.secret_map["session_token"].IsNull() ? "" : kv_secret.secret_map["session_token"].GetValue<string>());
-	auto signer = make_uniq<Aws::Client::AWSAuthV4Signer>(provider, service.c_str(), region.c_str());
-
-	signer->SignRequest(*req);
-	std::shared_ptr<Aws::Http::HttpResponse> res = MyHttpClient->MakeRequest(req);
-	Aws::Http::HttpResponseCode resCode = res->GetResponseCode();
-	DUCKDB_LOG_DEBUG(context, "iceberg.Catalog.Aws.HTTPRequest",
-	                 "GET %s (response %d) (signed with key_id '%s' for service '%s', in region '%s')",
-	                 uri.GetURIString(), resCode, kv_secret.secret_map["key_id"].GetValue<string>(), service.c_str(),
-	                 region.c_str());
-	if (resCode == Aws::Http::HttpResponseCode::OK) {
-		Aws::StringStream resBody;
-		resBody << res->GetResponseBody().rdbuf();
-		return resBody.str();
-	} else {
-		Aws::StringStream resBody;
-		resBody << res->GetResponseBody().rdbuf();
-		throw HTTPException(StringUtil::Format("Failed to query %s, http error %d thrown. Message: %s",
-		                                       req->GetUri().GetURIString(true), res->GetResponseCode(),
-		                                       resBody.str()));
-	}
+string &APIUtils::GetCURLCertPath() {
+	static string cert_path = "";
+	return cert_path;
```

Comment:
Btw, I just noticed we have a bit of complexity around the curl cert path, by changing it to a singleton with a static lifetime we could also change it to:
```c++
static string cert_path = SelectCURLCertPath();
return cert_path;
```

Because of the `static` lifetime, this guarantees that `SelectCURLCertPath()` is only called once, to initialize the value